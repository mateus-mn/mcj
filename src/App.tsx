import { QueryClient, QueryClientProvider } from 'react-query';
import ToastAlert from './components/ToastAlert';
import InicialPage from './pages/Inicial';

const App = () => {
	// seta o query Client para conseguir usar o React Query
	const queryClient = new QueryClient();

	return (
		<>
			<QueryClientProvider client={queryClient}>
				<InicialPage />
			</QueryClientProvider>

			{/* Inclui o toast que irá exibir todas as mensagens de alerta, erro, sucesso... nas páginas */}
			<ToastAlert />
		</>
	);
};

export default App;
