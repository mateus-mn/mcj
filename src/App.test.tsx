import { render } from '@testing-library/react';
import App from './App';

test('Teste do snapshot da página', () => {
	// renderiza o elemento
	const { container } = render(<App />);

	// espera-se que o snapshot seja o mesmo da página
	expect(container).toMatchSnapshot();
});
