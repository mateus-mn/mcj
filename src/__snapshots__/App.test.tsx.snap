// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do componente principal Teste de snapshot 1`] = `
<div>
  <div
    class="text-center"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="visually-hidden"
      >
         Carregando... 
      </span>
    </div>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
